beacon_node/http_api/src/attestation_rewards.rs

```diff
@@ -0,0 +1,260 @@
+            .safe_div(spec.effective_balance_increment)
+            .map_err(|e| custom_not_found(format!("Unable to get active_increments! {:?}", e)))?;
+
+        let base_reward_per_increment = BaseRewardPerIncrement::new(total_active_balance, spec)
+            .map_err(|e| {
+                custom_not_found(format!("Unable to get base_reward_per_increment! {:?}", e))
+            })?;
+
+        for effective_balance_eth in 0..=32 {
+            let base_reward = get_base_reward(
+                &state,
+                effective_balance_eth,
+                base_reward_per_increment,
+                spec,
+            );
+
+            let base_reward = base_reward.map_err(|e| {
+                warp_utils::reject::custom_not_found(format!("Unable to get base_reward! {:?}", e))
+            })?;
+
+            let reward_numerator = base_reward
+                .safe_mul(weight)
+                .and_then(|reward_numerator| {
+                    reward_numerator.safe_mul(unslashed_participating_increments)
+                })
+                .map_err(|_| {
+                    warp_utils::reject::custom_server_error(
+                        "Unable to calculate reward numerator".to_owned(),
+                    )
+                })?;
+
+            let ideal_reward = reward_numerator
+                .safe_div(active_increments)
+                .and_then(|ideal_reward| ideal_reward.safe_div(WEIGHT_DENOMINATOR))
+                .map_err(|_| {
+                    warp_utils::reject::custom_server_error(
+                        "Unable to calculate ideal_reward".to_owned(),
+                    )
+                })?;
+
+            if !state.is_in_inactivity_leak(previous_epoch, spec) {
+                ideal_rewards_hashmap.insert((*flag_index, effective_balance_eth), ideal_reward);
+            } else {
+                ideal_rewards_hashmap.insert((*flag_index, effective_balance_eth), 0);
+            }
+        }
+    }
+
+    let ideal_rewards: Vec<IdealAttestationRewards> = ideal_rewards_hashmap
+        .iter()
+        .fold(
+            HashMap::new(),
+            |mut acc, ((_flag_index, effective_balance_eth), ideal_reward)| {
+                let entry =
+                    acc.entry(*effective_balance_eth as u32)
+                        .or_insert(IdealAttestationRewards {
+                            effective_balance: *effective_balance_eth as u64,
+                            head: 0,
+                            target: 0,
+                            source: 0,
+                        });
+                match flag_index {
+                    TIMELY_SOURCE_FLAG_INDEX => entry.source += *ideal_reward,
+                    TIMELY_TARGET_FLAG_INDEX => entry.target += *ideal_reward,
+                    TIMELY_HEAD_FLAG_INDEX => entry.head += *ideal_reward,
+                    _ => {}
+                }
+                acc
+            },
+        )
+        .into_values()
+        .collect();
```

Comment:
I think we should do this conversion from `HashMap` to `Vec` right at the very end so that we can utilise the map when calculating rewards for individual validators below.

This will avoid the `.find()` that you do below, for which a `.get()` can be used instead.